l10n_fr_siren_lookup/wizard/siren_wizard.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+# © 2018 Le Filament (<http://www.le-filament.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import json
+import requests
+
+from odoo import api, fields, models
+
+URL = "https://data.opendatasoft.com/api/records/1.0/search/?dataset=sirene%40public&q={request}&rows=100"
+
+class SirenWizard(models.TransientModel):
+    _name = 'siren.wizard'
+    _description = 'Get values from companies'
+
+
+    ## Default functions
+    @api.model
+    def _default_name(self):
+        return self.env['res.partner'].browse(self.env.context.get('active_id')).name
+
+    @api.model
+    def _default_partner(self):
+        return self.env.context.get('active_id')
+
+    ## Fields
+    name = fields.Char(string='Company', default=_default_name)
+    line_ids = fields.One2many('siren.wizard.line', 'wizard_id', string="Results",)
+    partner_id = fields.Integer('Partner', default=_default_partner)
+
+
+    ## Action
+    @api.model
+    def _prepare_partner_from_data(self, data):
+        return {
+            'street': data.get('l4_normalisee', False),
+            'zip': data.get('codpos', False),
+            'city': data.get('libcom', False),
+            'siren': data.get('siren', False),
+            'siret': data.get('siret', False),
+            'category': data.get('categorie', False),
+            'creation_date': data.get('dcret', False),
+            'ape': data.get('apen700', False),
+            'ape_label': data.get('libapet', False),
+            'legal_type': data.get('libnj', False),
+            'staff': data.get('efetcent', 0),
+        }
+
+    def get_lines(self):
+        # Get request
+        r = requests.get(URL.format(request=self.name))
+        # Serialization request to JSON
+        companies = r.json()
+        # Unlink all company lines
+        self.line_ids.unlink()
+        # Fill new company lines
+        for company in companies['records']:
+            new_company = self.line_ids.create({
```

Comment:
I really think you don't need to create records here.
Just put all dict of values in a list and set the line_ids -> self.line_ids = [(6, False, list_of_dict)]
And you'll don't need anymore to do the unlink() before (because nothing is created in database)

Reply:
Thank you @apollet I have been trying to do what you suggest but I cannot get it to work !

I tried : 
```suggestion
        r = requests.get(URL.format(request=self.name))
        lines = []
        for line in r.json()['records']:
            lines.append(line['fields'])
        self.line_ids = [(6, False, lines)]
```

and 
```suggestion
        r = requests.get(URL.format(request=self.name))
        self.line_ids = [(6, False, r.json()['records'])]
```

I get the same TypeError: TypeError("unhashable type: 'dict'",)
I think I am missing something here... Any idea ?

Reply:
> I really think you don't need to create records here.

I don't understand why. creating a transient record onclicking on a button is a valid way. don't you think ?

Possible improvement : 
```
res = self._prepare_partner_from_data(company['fields'])
res['wizard_id'] = self.id
self.line_ids.create(res)
```

-> you replace a create + write by a single create. (faster)

+ moving ``'name': company['fields']['l1_normalisee']`` in the prepare.

Reply:
Sorry I was wrong, you have to use the "(0, 0,  { values })" thing.
Try this:

```
def get_lines(self):
    # Get request
    r = requests.get(URL.format(request=self.name))
    # Serialization request to JSON
    companies = r.json()
    # Fill new company lines
    companies_vals = []
    for company in companies['records']:
        res = self._prepare_partner_from_data(company['fields'])
        companies_vals.append((0, 0, res))
    self.line_ids = companies_vals
    return {"type": "ir.actions.do_nothing", } 
```

@legalsylvain Why create records (use database, slowdown user experience) when you don't have to ? :)

Reply:
AFAIK, ``self.line_ids = companies_vals`` will call write function of wizard and then create function of wizard line

```
(0, _, values)
    adds a new record created from the provided value dict.
```

Réf : https://www.odoo.com/documentation/12.0/reference/orm.html